packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -529,26 +525,107 @@ class RenderListWheelViewport
+  /// instantiate by the childManager.
   @override
   void performLayout() {
-    double currentOffset = 0.0;
-    RenderBox child = firstChild;
     final BoxConstraints innerConstraints =
-        constraints.copyWith(
-          minHeight: _itemExtent,
-          maxHeight: _itemExtent,
-          minWidth: 0.0,
-        );
-    while (child != null) {
+    constraints.copyWith(
+      minHeight: _itemExtent,
+      maxHeight: _itemExtent,
+      minWidth: 0.0,
+    );
+
+    int targetFirstIndex = offsetToIndex(offset.pixels - size.height / 2 - _itemExtent / 2);
+    int targetLastIndex = offsetToIndex(offset.pixels + size.height / 2 - _itemExtent / 2);
+
+    if(childManager.childCount != null) {
+      targetFirstIndex = targetFirstIndex.clamp(0, childManager.childCount - 1);
+      targetLastIndex = targetLastIndex.clamp(0, childManager.childCount - 1);
+    }
+
+    if(firstChild == null) {
+      _createChild(targetFirstIndex, after: null);
+      RenderBox child = firstChild;
       child.layout(innerConstraints, parentUsesSize: true);
       final ListWheelParentData childParentData = child.parentData;
-      // Centers the child in the cross axis. Consider making it configurable.
       final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
-      childParentData.offset = new Offset(crossPosition, currentOffset);
-      currentOffset += _itemExtent;
+      childParentData.offset = new Offset(crossPosition, indexToOffset(targetFirstIndex));
+    }
+
+    int currentFirstIndex = indexOf(firstChild);
+    int currentLastIndex = indexOf(lastChild);
+
+    // Remove all unnecessary children.
+    while(currentFirstIndex < targetFirstIndex) {
+      _destroyChild(firstChild);
+      currentFirstIndex++;
+    }
+
+    while(currentLastIndex > targetLastIndex) {
+      _destroyChild(lastChild);
+      currentLastIndex--;
+    }
+
+    // Relayout all active children.
+    RenderBox child = firstChild;
+    while(child != null) {
+      child.layout(innerConstraints, parentUsesSize: true);
       child = childAfter(child);
     }
 
+    // Spawning new children that are actually visible but not in child list.
+    while(currentFirstIndex > targetFirstIndex) {
+      _createChild(--currentFirstIndex, after: null);
+      child = firstChild;
+      child.layout(innerConstraints, parentUsesSize: true);
+      final ListWheelParentData childParentData = child.parentData;
+      final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
+      childParentData.offset = new Offset(crossPosition, indexToOffset(currentFirstIndex));
+    }
+    while(currentLastIndex < targetLastIndex) {
+      _createChild(++currentLastIndex, after: lastChild);
+      child = lastChild;
+      child.layout(innerConstraints, parentUsesSize: true);
+      final ListWheelParentData childParentData = child.parentData;
+      final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
```

Comment:
This is repeated in a bunch of places. Just create a private method for `setChildOffsetToIndex(RenderBox child, int index)`

Reply:
done.